diffsims/reciprocal_lattice_point/structure_factor.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2020 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+from diffpy.structure.symmetryutilities import (
+    expandPosition,
+    SymmetryConstraints,
+)
+import numpy as np
+
+from diffsims.reciprocal_lattice_point.atomic_scattering_parameters import (
+    get_atomic_scattering_parameters,
+    get_element_id_from_string,
+)
+
+
+def get_atomic_scattering_factor(atom, scattering_parameter):
+    """Return the atomic scattering factor f for a certain atom and
+    scattering parameter.
+
+    Parameters
+    ----------
+    atom : diffpy.structure.Atom
+        Atom with element type, Debye-Waller factor and occupancy number.
+    scattering_parameter : float
+        The scattering parameter s for these Miller indices describing
+        the crystal plane in which the atom lies.
+
+    Returns
+    -------
+    f : float
+        Scattering factor for this atom on this plane.
+    """
+    # Get the atomic scattering parameters
+    element_id = get_element_id_from_string(atom.element)
+    a, b = get_atomic_scattering_parameters(element_id)
+
+    # Get the scattering parameter squared
+    s2 = scattering_parameter ** 2
+
+    # Get the atomic scattering factor
+    f = element_id - (41.78214 * s2 * np.sum(a * np.exp(-b * s2)))
+
+    # Correct for occupancy and the Debye-Waller factor
+    dw_factor = np.exp(-atom.Bisoequiv * s2)
+    f *= atom.occupancy * dw_factor
+
+    return f
+
+
+def find_asymmetric_positions(positions, space_group):
+    """Return the asymmetric atom positions among a set of positions
+    when considering symmetry operations defined by a space group.
+
+    Parameters
+    ----------
+    positions : list
+        A list of cartesian atom positions.
+    space_group : diffpy.structure.spacegroups.SpaceGroup
+        Space group describing the symmetry operations.
+
+    Returns
+    -------
+    np.ndarray
+        Asymmetric atom positions.
+    """
+    asymmetric_positions = SymmetryConstraints(space_group, positions).corepos
+    return [
+        np.array([np.allclose(xyz, asym_xyz) for xyz in positions])
+        for asym_xyz in asymmetric_positions
+    ][0]
+
+
+def get_xray_structure_factor(phase, hkl, scattering_parameter):
+    """Assumes structure's lattice parameters and Debye-Waller factors
```

Comment:
Considering my comments yesterday were incorrect, I want to postpone a `radiation` attribute.

Reply:
I'm a little confused about this - as far as I can see the only parameterization of the atomic scattering factor that you've put in this PR is the Doyle-Turner one? That 1968 paper considered both X-rays and electrons.

So you mean that `get_xray_structure_factor` is getting the X-ray version of the Doyle-Turner parameterization and the `get_doyleturner_structure_factor` is getting the electron version of the Doyle-Turner parameterization using the Mott formula?

Where is the reference to `kinematical` coming from here? I think the Doyle-Turner parameterization you're using is an evaluation of the atomic scattering factor within the first Born approximation (i.e. kinematical) whether you're taking the X-ray version or the electron version?

To me the atomic scattering factor is just defined as being in the first Born approximation and this makes sense because that quantity is relevant both in kinematical theories of electron diffraction and dynamical theories of electron diffraction from crystals. See e.g. https://iopscience.iop.org/article/10.1088/0034-4885/42/11/002/pdf?casa_token=V2KN7WTLBSoAAAAA:VfkDr5GYF5qm4izwAFbAYgOcTwDVuOEBOjmSRk1QNrnfqEdXEEsrTRUf5HWCXVYtrL07fuQgNyLE

Reply:
I must admit that I've adapted the Doyle-Turner structure factor calculation from EMsoft (https://github.com/pyxem/diffsims/pull/111/files#diff-97bd368539d6d77fc0a03e1bc2c8691bR103). I've made this fact clear by referencing EMsoft in relevant docstrings. The atomic scattering factors are the same, for both methods, yes. The "kinematical" comes from the statement in EMsoft comments here: https://github.com/EMsoft-org/EMsoft/blob/develop/Source/EMsoftLib/diffraction.f90#L384.